Missing source location in error message from kprove

Reproduce:
Go to `k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic`
in `main.k`, remove the line `imports K-REFLECTION`.
Run `make -B KOMPILE_BACKEND=haskell`

Output:

```
/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/bin/kompile  --backend haskell  main.k -d .
[Warning] Compiler: Symbol '<c>' defined but not used. Add the 'unused'
attribute if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/./main.k)
	Location(22,15,27,5)
[Warning] Compiler: Symbol 'check_MAIN_Pgm' defined but not used. Add the
'unused' attribute if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/./main.k)
	Location(59,16,59,22)
[Warning] Compiler: Variable 'M' defined but not used. Prefix variable name
with underscore if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/imap.k)
	Location(11,10,11,11)
[Warning] Compiler: Variable 'V' defined but not used. Prefix variable name
with underscore if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/imap.k)
	Location(12,20,12,21)
/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/bin/kprove main-spec.k --smt-prelude imap.smt2  -d . | diff - main-spec.k.out
kore-exec: [2020-09-17 13:16:29.235035322] Error (ErrorException):
    Error:
      module 'MAIN':
      axiom declaration:
        Hooked sorts may not have subsorts.
    CallStack (from HasCallStack):
      error, called at app/share/GlobalMain.hs:499:36 in main:GlobalMain
[Warning] Critical: Found variable _ on right hand side of rule, not bound on
left hand side. Did you mean "?_"?
Note: this warning will become an error in subsequent releases.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/././main-spec.k)
	Location(9,20,9,21)
[Warning] Critical: Found variable _ on right hand side of rule, not bound on
left hand side. Did you mean "?_"?
Note: this warning will become an error in subsequent releases.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/././main-spec.k)
	Location(21,24,21,25)
[Warning] Critical: Found variable _ on right hand side of rule, not bound on
left hand side. Did you mean "?_"?
Note: this warning will become an error in subsequent releases.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/././main-spec.k)
	Location(38,24,38,25)
[Warning] Compiler: Variable 'M' defined but not used. Prefix variable name
with underscore if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/imap.k)
	Location(11,10,11,11)
[Warning] Compiler: Variable 'V' defined but not used. Prefix variable name
with underscore if this is intentional.
	Source(/home/hjorthjort/wasm-semantics/dev/deps/k/k-distribution/tests/regression-new/bisimulation/examples/spec-refinement/basic/imap.k)
	Location(12,20,12,21)
[Error] Critical: Haskell Backend execution failed with code 1 and produced no
output.
0a1
> #True
../../../../../../include/kframework/ktest.mak:95: recipe for target 'main-spec.k' failed
make: *** [main-spec.k] Error 1
```

--- a/kore/app/share/GlobalMain.hs
+++ b/kore/app/share/GlobalMain.hs
@@ -118,6 +118,9 @@ import Kore.IndexedModule.IndexedModule
     ( VerifiedModule
     )
 import Kore.Log as Log
+import Kore.Log.ErrorParse
+    ( errorParse
+    )
 import Kore.Parser
     ( ParsedPattern
     , parseKoreDefinition
@@ -506,7 +509,7 @@ mainParse parser fileName = do
     parseResult <-
         clockSomething "Parsing the file" (parser fileName contents)
     case parseResult of
-        Left err         -> error err
+        Left err         -> errorParse err
         Right definition -> return definition
 
 type LoadedModule = VerifiedModule Attribute.Symbol

--- a/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
+++ b/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
@@ -163,20 +163,19 @@ verifyNoHookedSupersort indexedModule axiom subsorts = do
             . getSortAttributes indexedModule
             . Subsort.supersort
         hookedSubsort = find isHooked subsorts
-    for_ hookedSubsort
-        $ \s -> koreFail
-            $ unlines
-                [ "Hooked sorts may not have subsorts."
-                , "hooked sort:"
-                , show . unparse $ Subsort.supersort s
-                , "its subsort:"
-                , show . unparse $ Subsort.subsort s
-                , "Location in the Kore file:"
-                , show . prettyPrintAstLocation
-                    $ locationFromAst (Subsort.supersort s)
-                , "Location in the original K file: "
-                , show . pretty $ Attribute.sourceLocation axiom
-                ]
+    for_ hookedSubsort $ \sort ->
+        koreFail . unlines $
+            [ "Hooked sorts may not have subsorts."
+            , "Hooked sort:"
+            , show . unparse $ Subsort.supersort sort
+            , "Its subsort:"
+            , show . unparse $ Subsort.subsort sort
+            , "Location in the Kore file:"
+            , show . prettyPrintAstLocation
+                $ locationFromAst (Subsort.supersort sort)
+            , "Location in the original K file: "
+            , show . pretty $ Attribute.sourceLocation axiom
+            ]
 
 verifyAxiomAttributes
     :: forall error attrs

--- a/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
+++ b/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
@@ -32,6 +32,7 @@ import qualified Data.Functor.Foldable as Recursive
 import Kore.ASTVerifier.Error
 import qualified Kore.Attribute.Axiom as Attribute
     ( Axiom
+    , sourceLocation
     )
 import Kore.Attribute.Hook
 import Kore.Attribute.Overload
@@ -62,6 +63,7 @@ import Kore.Unparser
     ( unparse
     )
 import qualified Kore.Verified as Verified
+import Pretty
 
 parseAttributes :: MonadError (Error VerifyError) m => Attributes -> m Hook
 parseAttributes = Attribute.Parser.liftParser . Attribute.Parser.parseAttributes
@@ -146,9 +148,10 @@ verifyNoHookAttribute attributes = do
 verifyNoHookedSupersort
     :: MonadError (Error VerifyError) error
     => IndexedModule Verified.Pattern Attribute.Symbol attrs
+    -> Attribute.Axiom SymbolOrAlias VariableName
     -> [Subsort.Subsort]
     -> error ()
-verifyNoHookedSupersort indexedModule subsorts= do
+verifyNoHookedSupersort indexedModule axiom subsorts = do
     let isHooked =
             getHasDomainValues . hasDomainValues
             . getSortAttributes indexedModule
@@ -162,6 +165,8 @@ verifyNoHookedSupersort indexedModule subsorts= do
                 , show . unparse $ Subsort.supersort s
                 , "its subsort:"
                 , show . unparse $ Subsort.subsort s
+                , "Location: "
+                , show . pretty $ Attribute.sourceLocation axiom
                 ]
 
 verifyAxiomAttributes
@@ -173,7 +178,7 @@ verifyAxiomAttributes
 verifyAxiomAttributes indexedModule axiom = do
     let overload = axiom Lens.^. field @"overload"
         subsorts = getSubsorts (axiom Lens.^. field @"subsorts")
-    verifyNoHookedSupersort indexedModule subsorts
+    verifyNoHookedSupersort indexedModule axiom subsorts
     case getOverload overload of
             Nothing -> do
                 let newOverload = Overload Nothing

--- a/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
+++ b/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
@@ -42,6 +42,9 @@ import Kore.Attribute.Sort
 import Kore.Attribute.Sort.HasDomainValues
 import Kore.Attribute.Subsort as Subsort
 
+import Kore.AST.AstWithLocation
+    ( locationFromAst
+    )
 import qualified Kore.Attribute.Parser as Attribute.Parser
 import qualified Kore.Attribute.Symbol as Attribute
 import Kore.Error
@@ -55,6 +58,9 @@ import Kore.Syntax.Application
     ( SymbolOrAlias (..)
     )
 import Kore.Syntax.Definition
+import Kore.Syntax.Id
+    ( prettyPrintAstLocation
+    )
 import Kore.Syntax.Pattern
 import Kore.Syntax.Variable
     ( VariableName (..)
@@ -165,7 +171,10 @@ verifyNoHookedSupersort indexedModule axiom subsorts = do
                 , show . unparse $ Subsort.supersort s
                 , "its subsort:"
                 , show . unparse $ Subsort.subsort s
-                , "Location: "
+                , "Location in the Kore file:"
+                , show . prettyPrintAstLocation
+                    $ locationFromAst (Subsort.supersort s)
+                , "Location in the original K file: "
                 , show . pretty $ Attribute.sourceLocation axiom
                 ]
 

--- a/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
+++ b/kore/src/Kore/ASTVerifier/AttributesVerifier.hs
@@ -19,6 +19,10 @@ module Kore.ASTVerifier.AttributesVerifier
     ) where
 
 import qualified Control.Lens as Lens
+import Data.Foldable
+    ( find
+    , for_
+    )
 import Data.Generics.Product
 import Prelude.Kore
 
@@ -53,6 +57,9 @@ import Kore.Syntax.Definition
 import Kore.Syntax.Pattern
 import Kore.Syntax.Variable
     ( VariableName (..)
+    )
+import Kore.Unparser
+    ( unparse
     )
 import qualified Kore.Verified as Verified
 
@@ -138,11 +145,24 @@ verifyNoHookAttribute attributes = do
 
 verifyNoHookedSupersort
     :: MonadError (Error VerifyError) error
-    => [Kore.Attribute.Sort.Sort]
+    => IndexedModule Verified.Pattern Attribute.Symbol attrs
+    -> [Subsort.Subsort]
     -> error ()
-verifyNoHookedSupersort supersortsAtts = do
-    let isHooked = getHasDomainValues . hasDomainValues <$> supersortsAtts
-    when (or isHooked) $ koreFail "Hooked sorts may not have subsorts."
+verifyNoHookedSupersort indexedModule subsorts= do
+    let isHooked =
+            getHasDomainValues . hasDomainValues
+            . getSortAttributes indexedModule
+            . Subsort.supersort
+        hookedSubsort = find isHooked subsorts
+    for_ hookedSubsort
+        $ \s -> koreFail
+            $ unlines
+                [ "Hooked sorts may not have subsorts."
+                , "hooked sort:"
+                , show . unparse $ Subsort.supersort s
+                , "its subsort:"
+                , show . unparse $ Subsort.subsort s
+                ]
 
 verifyAxiomAttributes
     :: forall error attrs
@@ -152,9 +172,8 @@ verifyAxiomAttributes
     -> error (Attribute.Axiom Internal.Symbol.Symbol VariableName)
 verifyAxiomAttributes indexedModule axiom = do
     let overload = axiom Lens.^. field @"overload"
-        supersorts = Subsort.supersort <$> getSubsorts (axiom Lens.^. field @"subsorts")
-        supersortsAtts = getSortAttributes indexedModule <$> supersorts
-    verifyNoHookedSupersort supersortsAtts
+        subsorts = getSubsorts (axiom Lens.^. field @"subsorts")
+    verifyNoHookedSupersort indexedModule subsorts
     case getOverload overload of
             Nothing -> do
                 let newOverload = Overload Nothing

--- a/kore/src/Kore/Log/ErrorParse.hs
+++ b/kore/src/Kore/Log/ErrorParse.hs
@@ -1,0 +1,40 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+
+module Kore.Log.ErrorParse
+    ( ErrorParse (..)
+    , errorParse
+    ) where
+
+import Prelude.Kore
+
+import Control.Monad.Catch
+    ( Exception (..)
+    , MonadThrow
+    , throwM
+    )
+import Pretty
+
+import Log
+
+newtype ErrorParse = ErrorParse { message :: String }
+    deriving Show
+
+instance Exception ErrorParse where
+    toException = toException . SomeEntry
+    fromException exn =
+        fromException exn >>= fromEntry
+
+instance Pretty ErrorParse where
+    pretty ErrorParse { message } =
+        Pretty.pretty message
+
+instance Entry ErrorParse where
+    entrySeverity _ = Error
+
+errorParse :: MonadThrow log => String -> log a
+errorParse message =
+    throwM ErrorParse { message }

--- a/kore/src/Kore/Log/Registry.hs
+++ b/kore/src/Kore/Log/Registry.hs
@@ -71,6 +71,9 @@ import Kore.Log.ErrorDecidePredicateUnknown
 import Kore.Log.ErrorException
     ( ErrorException
     )
+import Kore.Log.ErrorParse
+    ( ErrorParse
+    )
 import Kore.Log.ErrorRewriteLoop
     ( ErrorRewriteLoop
     )
@@ -151,6 +154,7 @@ entryHelpDocs :: [Pretty.Doc ()]
     , mk $ Proxy @DebugSubstitutionSimplifier
     , mk $ Proxy @ErrorBottomTotalFunction
     , mk $ Proxy @ErrorDecidePredicateUnknown
+    , mk $ Proxy @ErrorParse
     , mk $ Proxy @WarnFunctionWithoutEvaluators
     , mk $ Proxy @WarnSymbolSMTRepresentation
     , mk $ Proxy @WarnStuckClaimState

--- a/test/no-injection-into-hooked/test.k.verify.out.golden
+++ b/test/no-injection-into-hooked/test.k.verify.out.golden
@@ -2,3 +2,12 @@ Error:
   module 'TEST':
   axiom declaration:
     Hooked sorts may not have subsorts.
+    Hooked sort:
+    SortInt{}
+    Its subsort:
+    SortS{}
+    Location in the Kore file:
+    "./test-kompiled/definition.kore 209:136"
+    Location in the original K file: 
+    
+    

